src/tools/pub/publishing.md

```diff
@@ -319,13 +319,59 @@ pub.dev updates the package listing to display
 `2.0.0` as the stable version of the package.
 
 If all of the conditions at the beginning of this section are true,
-then you can ignore the following warning from `dart pub publish`: 
+then you can ignore the following warning from `dart pub publish`:
 
    *"Packages with an SDK constraint on a pre-release of the Dart SDK should
    themselves be published as a pre-release version. If this package needs Dart
    version 2.12.0-0, consider publishing the package as a pre-release
    instead."*
 
+
+## Retracting a package version {#retract}
+
+To prevent new package consumers from adopting a recently
+published version of your package, you can retract that package version
+within 7 days of publication.
+The retracted version can be restored again within 7 days of retraction.
+
+A retracted package version isn't deleted. It appears in the version
+listing of the package on pub.dev in the **Retracted versions** section. Also, the
+detailed view of that package version has a **RETRACTED** badge.
+
+Before retracting a package,
+consider publishing a new version instead.
+Retracting a package causes churn and can have a negative impact on package users.
+
+For example, if your package version contains a minor bug, then retraction is
```

Comment:
I'd argue that the **best** example of when this should be used is a missing or overly weak dependency constraint that allows versions of SDKs or other packages that won't work. In this case, publishing a new version won't help because folks can still "solve" to the broken version with the too-loose constraints.

Reply:
It actually says that for minor bugs, it's "probably not needed". And then later says: "if you pulish a major version, or a version with missing or wrong dependency constraints then retraction ...".

Should we start by stating the positive case, and focusing on the "too-loose constraints" problem.
(Missing dependency constraints is actually the extreme variant of that).

Reply:
I like that idea. What would that edit look like?

Reply:
Actually, I need this PR to land as the CVE I'm about to publish, so can we handle that re-write in a follow-up PR?

Reply:
Filed in https://github.com/dart-lang/site-www/pull/3707